src/config.rs

```diff
@@ -117,41 +161,57 @@ macro_rules! create_config {
-    where_indent: BlockIndentStyle, // Visual will be treated like Tabbed
-    where_layout: ListTactic,
-    where_pred_indent: BlockIndentStyle,
-    generics_indent: BlockIndentStyle,
-    struct_trailing_comma: SeparatorTactic,
-    struct_lit_trailing_comma: SeparatorTactic,
-    struct_lit_style: StructLitStyle,
-    struct_lit_multiline_style: MultilineStyle,
-    enum_trailing_comma: bool,
-    report_todo: ReportTactic,
-    report_fixme: ReportTactic,
-    reorder_imports: bool, // Alphabetically, case sensitive.
-    single_line_if_else: bool,
-    format_strings: bool,
-    chains_overflow_last: bool,
-    take_source_hints: bool, // Retain some formatting characteristics from
-                             // the source code.
+    max_width: usize, "Maximum width of each line",
+    ideal_width: usize, "Ideal width of each line",
+    leeway: usize, "Leeway of line width",
+    tab_spaces: usize, "Number of spaces per tab",
+    newline_style: NewlineStyle, "Unix or Windows line endings",
+    fn_brace_style: BraceStyle, "Brace style for functions",
+    fn_return_indent: ReturnIndent, "Location of return type in function declaration",
+    fn_args_paren_newline: bool, "If function argument parenthases goes on a newline",
+    fn_args_density: Density, "Argument density in functions",
+    fn_args_layout: StructLitStyle, "Layout of function arguments",
+    fn_arg_indent: BlockIndentStyle, "Indent on function arguments",
+    where_density: Density, "Density of a where clause", // Should we at least try to put the where clause on the same line as
+                                                         // the rest of the function decl?
+    where_indent: BlockIndentStyle, "Indentation of a where clause", // Visual will be treated like Tabbed
+    where_layout: ListTactic, "Element layout inside a where clause",
+    where_pred_indent: BlockIndentStyle, "Indentation style of a where predicate",
+    generics_indent: BlockIndentStyle, "Indentation of generics",
+    struct_trailing_comma: SeparatorTactic, "If there is a trailing comma on structs",
+    struct_lit_trailing_comma: SeparatorTactic, "If there is a trailing comma on literal structs",
+    struct_lit_style: StructLitStyle, "Style of struct definition",
+    struct_lit_multiline_style: MultilineStyle, "Multilline style on literal structs",
+    enum_trailing_comma: bool, "Put a trailing comma on enum declarations",
+    report_todo: ReportTactic, "Report all occurences of TODO in source file comments",
+    report_fixme: ReportTactic, "Report all occurences of FIXME in source file comments",
+    reorder_imports: bool, "Reorder import statements alphabetically", // Alphabetically, case sensitive.
+    single_line_if_else: bool, "Put else on same line as closing brace for if statements",
+    format_strings: bool, "Format string literals, or leave as is",
+    chains_overflow_last: bool, "chains overflow last",
```

Comment:
That's getting close. What about "Allow last call in method chain to break the line"?